lib/action_client/base.rb

```diff
@@ -3,34 +3,55 @@ class Base < AbstractController::Base
     include AbstractController::Rendering
     include ActionView::Layouts
 
-    cattr_accessor :defaults,
+    class_attribute :defaults,
       instance_accessor: true,
       default: ActiveSupport::OrderedOptions.new
 
+    class_attribute :middleware,
+      instance_accessor: true
+
+    class_attribute :after_actions,
+      instance_accessor: true,
+      default: []
+
     class << self
+      def inherited(descendant)
+        descendant.middleware = Rails.configuration.action_client.middleware.dup
+        descendant.after_actions = after_actions.dup
+      end
+
       def default(options)
         options.each do |key, value|
           defaults[key] = value
         end
       end
 
+      def after_action(&block)
+        after_actions.append(block)
+      end
+
+      def rescue_from(*rescued_status_codes, &block)
```

Comment:
I settled on `rescue_from`, since it's conceptually and syntactically close to [ActiveSupport's `rescue_from`](https://api.rubyonrails.org/classes/ActiveSupport/Rescuable/ClassMethods.html#method-i-rescue_from). 

Despite that fact, I feel like the examples in this commit aren't about "rescuing" at all, and are all about causing and raising exceptions, so I'm very open to suggestions on more appropriate name.

Reply:
Given how similar this is to an `after_action`, I wonder if an argument would be more appropriate than a separate method? e.g. `after_action with_status: 400 { |_| raise "Bad request!" }`

This would be kind of similar to Rails' `after_action if: :whatever { … }`